staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -877,8 +894,11 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 		r.retry.After(ctx, r, resp, err)
 
 		done := func() bool {
-			defer readAndCloseResponseBody(resp)
+			// make sure we always drain and close
+			closer := NewResponseCloser(resp)
+			defer closer.DrainAndClose()
 
+			respCopy := shallowCloneHTTPResponse(resp)
```

Comment:
if we replace resp.Body internally in DrainAndClose(), I don't think we need to do this